azure/scope/managedcontrolplane.go

```diff
@@ -501,6 +501,7 @@ func (s *ManagedControlPlaneScope) GetAgentPoolSpecs(ctx context.Context) ([]azu
 			OSDiskSizeGB:      0,
 			Mode:              pool.Spec.Mode,
 			AvailabilityZones: pool.Spec.AvailabilityZones,
+			OsDiskType:        pool.Spec.OsDiskType,
```

Comment:
@zmalik @CecileRobertMichon moved these assignments here (instead of lower in a "if not empty" condition) in response to this PR:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1945